apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-confirm.component.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { FormControl, FormGroup } from '@angular/forms';
+import { MatBottomSheetRef } from '@angular/material/bottom-sheet';
+import { MatDialog } from '@angular/material/dialog';
+import { createRoutingFactory, Spectator } from '@ngneat/spectator/jest';
+import { BookingFormService } from '@placeos/bookings';
+import {
+    Building,
+    BuildingLevel,
+    Desk,
+    OrganisationService,
+} from '@placeos/organisation';
+import { PlaceZone } from '@placeos/ts-client';
+import { NewDeskFlowConfirmComponent } from 'apps/workplace/src/app/book/new-desk-flow/new-desk-flow-confirm.component';
+
+describe('NewDeskFlowConfirmComponent', () => {
+    let spectator: Spectator<NewDeskFlowConfirmComponent>;
+    const createComponent = createRoutingFactory({
+        component: NewDeskFlowConfirmComponent,
+        providers: [
+            {
+                provide: BookingFormService,
+                useValue: {
+                    form: new FormGroup({
+                        date: new FormControl(new Date()),
```

Comment:
I feel like this form group is mostly implementation details and isn't necessary for the tests.
If you do need the correct form use the function from the utilities to generate it.

Reply:
Using generate form from utilities